### PR TITLE
Fix/segment triangle distance

### PR DESCRIPTION
**Pull request by panquez**

### CODE REVIEW

File: src/geode/geometry/distance.cpp [hunk 1]
```diff
@@ -749,6 +806,108 @@ namespace geode
...
+        // | Dot(E1, E1)  Dot(E1, E2) | | b[1] | = | Dot(E1, Y - V[0]) |
+        // | Dot(E1, E2)  Dot(E2, E2) | | b[2] |   | Dot(E2, Y - V[0]) |
+        // +-                        -+ +-    -+   +-                 -+
+        //
+        // and b[0] = 1 - b[1] - b[2].
+
+        // Test whether the line intersects triangle. If so, the squared
+        // distance is zero. The normal of the plane of the triangle does
+        // not have to be normalized to unit length.
+        const Vector3D E1{ triangle.vertices()[0], triangle.vertices()[1] };
```
Comment (BotellaA) [inline src/geode/geometry/distance.cpp:830]: extract `triangle.vertices()` at the begining?

File: src/geode/geometry/intersection.cpp [hunk 2]
```diff
@@ -382,10 +378,10 @@ namespace geode
                 auto result = line.origin() + line.direction() * seg_parameter;
 
                 CorrectnessInfo< Point3D > correctness;
-                double distance;
-                std::tie( distance, correctness.first.second ) =
-                    point_line_distance( result, line );
-                correctness.first.first = distance < global_epsilon;
+                auto distance = point_line_distance( result, line );
```
Comment (BotellaA) [inline src/geode/geometry/intersection.cpp:381]:
```suggestion
                const auto distance = point_line_distance( result, line );
```

File: src/geode/geometry/intersection.cpp [hunk 3]
```diff
@@ -769,19 +763,16 @@ namespace geode
         correctness.second.first = true;
         for( const auto r : LRange{ result.numIntersections } )
         {
-            double distance;
-            Point3D point;
-
             // distance to line
-            std::tie( distance, point ) =
-                point_line_distance( results[r], line );
+            auto distance = point_line_distance( results[r], line );
```
Comment (BotellaA) [inline src/geode/geometry/intersection.cpp:767]:
```suggestion
            const auto distance = point_line_distance( results[r], line );
```

**Review by BotellaA — APPROVED** [review]

**Review by francoisbonneau — APPROVED** [review]

**Pull request merged**